### PR TITLE
Turns a lot of math procs into defines

### PR DESCRIPTION
**Pull request by ninjanomnom**
Proc call overhead means we can't have nice things.

It's shit to be worried about proc call overhead when you just want to make some math a bit more readable.

### CODE REVIEW

File: code/__DEFINES/maths.dm [hunk 1]
```diff
@@ -1,234 +1,206 @@
...
+
+#define NUM_E 2.71828183
+#define NUM_SQRT2 1.41421356
+
+#define PI						3.1415
+#define SPEED_OF_LIGHT			3e8		//not exact but hey!
+#define SPEED_OF_LIGHT_SQ		9e+16
+#define INFINITY				1e31	//closer then enough
+
+//atmos
```
Comment (as334) [inline code/__DEFINES/maths.dm]: These can probably get moved to the atmospherics defines.

File: code/__DEFINES/atmospherics.dm [hunk 2]
```diff
@@ -1,3 +1,9 @@
+#define R_IDEAL_GAS_EQUATION	8.31	//kPa*L/(K*mol)
```
Comment (optimumtact) [inline code/__DEFINES/atmospherics.dm]: this is just a number

Comment (banthebantz) [reply code/__DEFINES/atmospherics.dm]: this is a constant not an equation lmao

File: code/__DEFINES/maths.dm [hunk 3]
```diff
@@ -1,234 +1,199 @@
...
+// Will filter out extra rotations and negative rotations
+// E.g: 540 becomes 180. -180 becomes 180.
+#define SIMPLIFY_DEGREES(degrees) ((degrees*SIGN(degrees))%360)
+
+// min is inclusive, max is exclusive
+#define WRAP(val, min, max) (val - ((round(val - min) / (max - min)) * (max - min)))
+
+#define NORM_ROT(rot) ((((rot % 360) + (rot - round(rot, 1))) >= 0) ? ((rot % 360) + (rot - round(rot, 1))) : (((rot % 360) + (rot - round(rot, 1))) + 360))
+
+/proc/get_angle_of_incidence(face_angle, angle_in, auto_normalize = TRUE)
```
Comment (nicbn) [inline code/__DEFINES/maths.dm]:
```
#define GET_ANGLE_OF_INCIDENCE(face_angle, angle_in) ( NORM_ROT(angle_in) - NORM_ROT(face_angle) )

/proc/get_normalized_angle_of_incidence(face_angle, angle_in)
	. = GET_ANGLE_OF_INCIDENCE(face_angle, angle_in)
	while(. < -90)
		. += 180
	while(. > 90)
		. -= 180
```

File: code/__DEFINES/maths.dm [hunk 4]
```diff
@@ -1,234 +1,199 @@
...
+			R2 = rand(-ACCURACY,ACCURACY)/ACCURACY
+			working = R1*R1 + R2*R2
+		while(working >= 1 || working==0)
+		working = sqrt(-2 * log(working) / working)
+		R1 *= working
+		gaussian_next = R2 * working
+	return (mean + stddev * R1)
+#undef ACCURACY
+
+/proc/mouse_angle_from_client(client/client)
```
Comment (nicbn) [inline code/__DEFINES/maths.dm]:
```
//Refer to http://www.byond.com/docs/ref/info.html#/client/var/view for mad maths
/proc/mouse_angle_from_client(client/c)
	var/screen_loc = params2list(c.mouseParams)["screen-loc"]
	if(screen_loc)
		var/list/screen_loc_params = splittext(screen_loc, ",")
		var/list/screen_loc_X = splittext(screen_loc_params[1],":")
		var/list/screen_loc_Y = splittext(screen_loc_params[2],":")
		var/halfscreen = round((c.view * 2 + 1) * world.icon_size / 2)
		return NORM_ROT(ATAN2((text2num(screen_loc_Y[1]) * 32 + text2num(screen_loc_Y[2]) - 32) - halfscreen + c.pixel_y, (text2num(screen_loc_X[1]) * 32 + text2num(screen_loc_X[2]) - 32) - halfscreen + c.pixel_x))
```

File: code/__DEFINES/maths.dm [hunk 5]
```diff
@@ -1,234 +1,199 @@
...
+
+#define INVERSE(x) (1/x)
+
+#define INVERSE_SQUARE(initial_strength,cur_distance,initial_distance) (initial_strength*(initial_distance**2/cur_distance**2))
+
+#define ISABOUTEQUAL(a, b, deviation) (deviation ? abs(a - b) <= deviation : abs(a - b) <= 0.1)
+
+#define ISEVEN(x) (x % 2 == 0)
+
+#define ISODD(x) (!ISEVEN(x))
```
Comment (nicbn) [inline code/__DEFINES/maths.dm]:
```
#define ISODD(x) (x % 2 != 0)
```

2 operations is faster than 3 operations

File: code/__DEFINES/atmospherics.dm (hunk 2 above)
Comment (duncathan) [reply code/__DEFINES/atmospherics.dm]: the define is poorly named, I'll rename it to `R_IDEAL_GAS_CONSTANT` after this pr

File: code/__DEFINES/maths.dm (hunk 4 above)
Comment (ninjanomnom) [reply code/__DEFINES/maths.dm]: too little benefit for the loss of readability

File: code/__DEFINES/maths.dm [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+#define DEFAULT(a, b) (a ? a : b)
+
+// Greatest Common Divisor - Euclid's algorithm
+/proc/Gcd(a, b)
+	return b ? Gcd(b, a % b) : a
+
+// Least Common Multiple
+#define Lcm(a, b) (abs(a) / Gcd(a, b) * abs(b))
+
+#define INVERSE(x) (1/x)
```
Comment (nicbn) [inline code/__DEFINES/maths.dm]: Does this define really get used anywhere in the code?

Comment (ninjanomnom) [reply code/__DEFINES/maths.dm]: no but it's fine to keep, we've got quite a few unused math helpers

**Comment by duncathan** [comment]
haha nice

**Pull request merged**